### PR TITLE
Fail if ROWID shape and chunks don't match associated dataset arrays

### LINKED ISSUE #238: Mismatched chunks between data_vars and ROWID gives weird results.
* dask-ms version: master
* Python version: 3.8.10
* Operating System: Ubuntu 20.04 LTS

### Description

While debugging some other code I ran into a situation in which dask-ms quietly accepts bad user input and then produces mysterious results. Specifically, if the chunking on `ROWID` does not mach the chunking in `row` on the `xds.data_vars`, dask-ms does not complain but the write goes wrong.

### Reproducer
```python
from daskms import xds_from_ms, xds_to_table
import dask
from numpy.testing import assert_array_equal


if __name__ == "__main__":

    ms_name = "C147_unflagged.MS"

    xdsl = xds_from_ms(
        ms_name,
        group_cols=["SCAN_NUMBER"],
        chunks={"row": -1},
        columns=["DATA"]
    )

    xds = xdsl[0]

    # NOTE: Remove this line to make this test pass.
    xds = xds.assign_coords({"ROWID": (("row",), xds.ROWID.data.rechunk(10000))})

    ref_data = xds.DATA.values  # Prior to write.

    xds = xds.rename({"DATA": "BROKEN_DATA"})

    writes = xds_to_table(xds, ms_name, columns=["BROKEN_DATA"])

    dask.compute(writes)

    xdsl = xds_from_ms(
        ms_name,
        group_cols=["SCAN_NUMBER"],
        chunks={"row": -1},
        columns=["BROKEN_DATA"]
    )

    xds = xdsl[0]

    upd_data = xds.BROKEN_DATA.values  # Post write.

    assert_array_equal(ref_data, upd_data)
```

### PR DESCRIPTION
**Pull request by sjperkins**
Closes #238 

- [x] Tests added / passed

  ```bash
  $ py.test -v -s daskms/tests
  ```

  If the pep8 tests fail, the quickest way to correct
  this is to run `autopep8` and then `flake8` and
  `pycodestyle` to fix the remaining issues.

  ```
  $ pip install -U autopep8 flake8 pycodestyle
  $ autopep8 -r -i daskms
  $ flake8 daskms
  $ pycodestyle daskms
  ```

- [x] Fully documented, including `HISTORY.rst` for all changes
      and one of the `docs/*-api.rst` files for new API

  To build the docs locally:

  ```
  pip install -r requirements.readthedocs.txt
  cd docs
  READTHEDOCS=True make html
  ```

### CODE REVIEW

**Comment by sjperkins** [comment]
Running this PR on the reproducer in #239  now results in the following error:

```bash
Traceback (most recent call last):
  File "tmp/repro.py", line 26, in <module>
    writes = xds_to_table(xds, ms_name, columns=["BROKEN_DATA"])
  File "/home/simon/code/dask-ms/daskms/dask_ms.py", line 96, in xds_to_table
    out_ds = write_datasets(table_name, xds, columns,
  File "/home/simon/code/dask-ms/daskms/writes.py", line 725, in write_datasets
    write_datasets = _write_datasets(table, tp, datasets, columns,
  File "/home/simon/code/dask-ms/daskms/writes.py", line 629, in _write_datasets
    raise ValueError(f"ROWID shape and chunking does not "
ValueError: ROWID shape and chunking does not match that of BROKEN_DATA
```

File: daskms/writes.py [hunk 1]
```diff
@@ -624,6 +624,11 @@ def _write_datasets(table, table_proxy, datasets, columns, descriptor,
 
             inlinable_arrays = [row_order]
 
+            if (row_order.shape[0] != array.shape[0] or
+                    row_order.chunks[0] != array.chunks[0]):
+                raise ValueError(f"ROWID shape and chunking does not "
```
Comment (JSKenyon) [inline daskms/writes.py:629]: Nitpick - "ROWID shape and/or chunking does not match that of {column}."

**Review by JSKenyon — APPROVED** [review]
Looks good to me, bar my nitpick.

**Comment by sjperkins** [comment]
Thanks for the review!

**Pull request merged**